albinos-cli/source/cli/cli.d

```diff
@@ -0,0 +1,123 @@
+module cli.cli;
+
+import core.exception : AssertError;
+import std.stdio;
+import std.string;
+import std.socket : SocketOSException;
+import std.getopt;
+import std.uni : isWhite;
+import std.process : executeShell;
+import std.file;
+import asdf;
+import protocol.protocol_type;
+import client.client;
+import cli.deserializer;
+
+class CLI
+{
+    this(string socket_path)
+    {
+        try
+        {
+            client_ = new Client(socket_path);
+        }
+        catch (SocketOSException e)
+        {
+            writeln(stderr, e.msg);
+            is_running_ = false;
+        }
+    }
+
+    private void load_config(string[] args) 
+    in
+    {
+        assert(args !is null, "args cannot be null");
+        assert(args.length == 2, "need 1 arguments");
+    }
+    body
+    {
+        string config_key;
+        getopt(args, "key", &config_key);
+        auto cfg = deserialize_to!config_load("../API_doc/json_recipes/config_load.json");
+        cfg.config_key = config_key;
+        writeln(cfg);
+    }
+
+    private config_create_answer create_config(string[] args)
+    in
+    {
+        assert(args !is null, "args cannot be null");
+        assert(args.length == 2, "need 1 arguments");
+    }
+    out (r)
+    {
+        assert(r.state == "SUCCESS", "create_config should success");
+        assert(!r.config_key.empty, "config_key should not be empty");
+        assert(!r.readonly_config_key.empty, "readonly_config_key should not be empty");
+    }
+    body
+    {
+        string config_name;
+        getopt(args, "name", &config_name);
+        auto cfg = deserialize_to!config_create("../API_doc/json_recipes/config_create.json");
+        cfg.config_name = config_name.strip("\"");
```

Comment:
I'm not sure this is the way to proceed. If you are expecting quotes, why not remove the first and last character?
If it varies, I think this would need to verify the input is well formed (ex: `"jack"robinson`)

Reply:
GetOpt when you do --arg="toto" keep the quote and the string receive is " "toto " " when we do --arg=toto we receive "toto"

Reply:
Could you add a unit test to confirme everything works on malformed inputs ?